### DISCUSSION TITLE
How to make a (pre-built) static library's contents accessible to a test?

### DISCUSSION THREAD

**Discussion by krispin-gradient** [ThrowTheSwitch/Ceedling#953] [Q&A] [answered]
I'm trying to get Ceedling (0.3x) to compile a test file, such that it links in a pre-built static library (.a file).

I can add a path to the _folder_ containing the pre-built static library, e.g:
```
  :libraries:
     - path/to/lib-folder
```
...and the executed command reflects this, e.g:
```
gcc ... -L path/to/lib-folder -o "test.out" -lm 
```
However the command that needs to execute is one _including the library file itself_, e.g:
```
gcc ... -L path/to/lib-folder -lmyLib.a -o "test.out" -lm 
```
Is there a way to configure Ceedling to yield a command including specific libraries, i.e. **-lmyLib.a**?

**Comment by mvandervoord** [comment, marked as answer]
You want to specify the path in paths and the library itself in the libraries section:

```
:paths:
  :libraries:
    - path/to/lib-folder    # Linker library search paths

:libraries:
  :test:
    - myLib.a
  :release: []
  :system:
    - m
  :flag: -l${1}  # Not actually necessary for gcc-conventions
```

**Comment by krispin-gradient** [comment]
Ah - thanks!
